Services/PaymentVault/PaymentDeviceVault.php

```diff
@@ -96,7 +96,14 @@ public function saveDeviceToVault(BasePaymentType $paymentType, string $deviceTy
 
         if ($deviceType === VaultedDeviceStruct::DEVICE_TYPE_PAYPAL &&
             $paymentType instanceof Paypal &&
-            $this->paypalAccountExists((int) $this->session->offsetGet('sUserId'), $paymentType->getEmail(), $addressHash)) {
+            (
+                $paymentType->getEmail() === null ||
+                $this->paypalAccountExists(
+                    (int) $this->session->offsetGet('sUserId'),
+                    $paymentType->getEmail(),
+                    $addressHash
+                )
```

Comment:
this seems to be done twice, is that correct?

Reply:
What exactly do you mean?

Reply:
This line in the condition of the if statement
`$this->paypalAccountExists((int) $this->session->offsetGet('sUserId'), $paymentType->getEmail(), $addressHash)`

And this in its body:
```
$this->paypalAccountExists(
                    (int) $this->session->offsetGet('sUserId'),
                    $paymentType->getEmail(),
                    $addressHash
                )
```

Reply:
Since this line is an "and" construct in the condition, it must be true when the method is entered.
So it will always be true in the "or" construct and does not need to be taken into account.

Reply:
The body of the if statement is only the `return`.  The first condition you've mentioned was removed, that's why it's highlighted red and prefixed with "-" 😉 

Basically, we're checking if `$payment->getEmail()` is null and if that's the case, `paypalAccountExists()` won't even be called because of the OR-operator. Because of that, the whole condition is true so we're directly returning to don't save a device without an email.  
If `$payment->getEmail()` is not null, `paypalAccountExists()` gets called and it depends on that if a device should be saved. I hope that clarifies the bug fix.

Reply:
Oje 🤦🏼 
Never mind

Reply:
maybe I am color blind